### DISCUSSION TITLE
Breaking change should have mahor version bump

### DISCUSSION THREAD

**Discussion by ChristianSch** [testcontainers/testcontainers-go#600] [General]
Hi,

I noticed that 0.15.0 includes a breaking change without bumping the major version. Please see here:
https://semver.org/

> MAJOR version when you make incompatible API changes

It would be very helpful to properly make use of that. Was there a reason not to do this?

While we're at it, would it be possible to have a CHANGELOG.md file, in addition to the github-bound format?

**Comment by mdelapenya** [comment]
Hi @ChristianSch, indeed. We are aware of semver and would love to adhere to it at the moment we get an stable, public API, reaching 1.0. For that reason we have broken the public APIs a few times and not bumped the major version of the project.

In that sense, we are very cautious about when to do a breaking change, and we try to notify it in the changelog. Some of the reasons that we have considered to break the API has been related to see changes rapidly into the project with no other option than to break the API.

On the other hand we understand that it could have caused some trouble in downstream consumers, and because of that we do try to be extremely conservative about it. I'd like to apologize if it broke you for some reasons.

About the CHANGELOG.md file, would like to understand the need to add it apart from the GH format. Is that an automated process that consumes it? We are open to add it if there is a valuable need on it.

**Comment by ChristianSch** [comment]
Thank you for the well-articulated response, I appreciate that! Being on 0.x.x and thus not doing major version bumps is a very good reason that I actually overlooked to be honest. I totally understand that you want to wait until the API feels rather stable. Although some projects never bump to v1, I hope this does 🤞 

CHANGELOG.md on the other hand is just convenience. If you don't update very version you have to click quite a lot to get a feeling for the breaking changes, as for the CHANGELOG.md file you can just scroll down. But I have no strong feeling either way, I was just kind of confused that it wasn't there.

**Reply by mdelapenya** [reply]
Never mind, I think having a CHANGELOG.md file is a very common practice, so we will discuss it internally. I say we as a testcontainers team, including all the language implementations: java, go, dotnet, etc.

**Reply by ChristianSch** [reply]
Awesome, thank you! Great feedback and discussion culture!

**Comment by mdelapenya** [comment]
I'm closing this now, as it's already been discussed. Thank you for your time
